Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,8 +380,14 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        X = BasicStats(self.data[:, self.attr_x], self.attr_x)
+        Y = BasicStats(self.data[:, self.attr_y], self.attr_y)
```

Comment:
The [discussion from #1070](https://github.com/biolab/orange3/pull/1070#discussion_r55517993) is still relevant, though. You can optimize this by computing basic stats only once, that is, when the data is connected.